grouper/fe/handlers/permission_request.py

```diff
@@ -0,0 +1,141 @@
+import json
+
+from tornado.web import HTTPError
+
+from grouper import permissions
+from grouper.audit import UserNotAuditor
+from grouper.fe.forms import PermissionRequestForm
+from grouper.fe.settings import settings
+from grouper.fe.util import Alert, GrouperHandler
+from grouper.models.group import Group
+from grouper.permissions import get_grantable_permissions, get_permission
+from grouper.user_group import get_groups_by_user
+
+
+def _build_form(request, data):
```

Comment:
Nit: This should be a method, since you're passing in the object anyway.
Nit: mypy type annotations on all new functions.

Reply:
Oh? My default is never to have methods if I don't need to — if I'm not mutating the object, I do a function. To match the conventions of the rest of the code, I'll re-method this.